docs/entities/livingentity.md

```diff
@@ -0,0 +1,272 @@
+
+At this point, all kinds of modifiers are available, allowing you to finely modify the damage amount. Be aware that things like armor damage are already done by this point.
+
+#### `LivingDamageEvent.Post`
+
+This event is called after the damage has been done, absorption has been reduced, the combat tracker has been updated, and stats and game events have been handled. It is not cancellable, as the attack has already happened. This event would commonly be used for post-attack effects. Note that the event is fired even if the damage amount is zero, so check that value accordingly if needed.
+
+If you are calling this on your own entity, you should consider overriding `ILivingEntityExtension#onDamageTaken()` instead. Unlike `LivingDamageEvent.Post`, this is only called if the damage is greater than zero.
+
+## Mob Effects
+
+_See [Mob Effects & Potions][mobeffects]._
+
+## Equipment
+
+_See [Containers on Entities][containers]._
+
+## Hierarchy
+
+Living entities have a complex class hierarchy. As mentioned before, there are three direct subclasses:
+
+```mermaid
+graph LR;
+    LivingEntity-->ArmorStand;
+    LivingEntity-->Mob;
+    LivingEntity-->Player;
+```
+
+Of these, `ArmorStand` has no subclasses (and is also the only non-abstract class), so we will focus on the class hierarchy of `Mob` and `Player`.
+
+### Hierarchy of `Mob`
+
+The class hierarchy of `Mob` looks as follows (red classes are `abstract`, blue classes are not):
+
+```mermaid
+graph LR;
+    Mob-->AmbientCreature;
+    AmbientCreature-->Bat;
+    Mob-->EnderDragon;
+    Mob-->FlyingMob;
+    FlyingMob-->Ghast;
+    FlyingMob-->Phantom;
+    Mob-->PathfinderMob;
+    PathfinderMob-->AbstractGolem;
+    AbstractGolem-->IronGolem;
+    AbstractGolem-->Shulker;
+    AbstractGolem-->SnowGolem;
+    PathfinderMob-->AgeableMob;
+    AgeableMob-->AbstractVillager;
+    AbstractVillager-->Villager;
+    AbstractVillager-->WanderingTrader;
+    AgeableMob-->AgeableWaterCreature;
+    AgeableWaterCreature-->Dolphin;
+    AgeableWaterCreature-->Squid;
+    Squid-->GlowSquid;
+    AgeableMob-->Animal;
+    PathfinderMob-->Allay;
+    PathfinderMob-->Monster;
+    PathfinderMob-->WaterAnimal;
+    WaterAnimal-->AbstractFish;
+    AbstractFish-->AbstractSchoolingFish;
+    AbstractSchoolingFish-->Cod;
+    AbstractSchoolingFish-->Salmon;
+    AbstractSchoolingFish-->TropicalFish;
+    AbstractFish-->Pufferfish;
+    AbstractFish-->Tadpole;
+    Mob-->Slime;
+    Slime-->MagmaCube;
+    
+    class Mob,AmbientCreature,FlyingMob,PathfinderMob,AbstractGolem,AgeableMob,AbstractVillager,AgeableWaterCreature,Animal,Monster,WaterAnimal,AbstractFish,AbstractSchoolingFish red;
+    class Bat,EnderDragon,Ghast,Phantom,IronGolem,Shulker,SnowGolem,Villager,WanderingTrader,Dolphin,Squid,GlowSquid,Allay,Cod,Salmon,TropicalFish,Pufferfish,Tadpole,Slime,MagmaCube blue;
```

Comment:
Feels a bit strange that this diagram is color-coded but the others on the page aren't -- maybe could document that somewhere, and possibly add something to the contributing guidelines to standardize it?